metronome/hotstuff/service/test/src/io/iohk/metronome/hotstuff/service/storage/BlockStorageProps.scala

```diff
@@ -238,6 +254,53 @@ object BlockStorageProps extends Properties("BlockStorage") {
       data.store.getPathFromRoot(nonExisting.id).isEmpty
   }
 
+  property("getPathFromAncestor") = forAll(
+    for {
+      prefix <- genNonEmptyBlockTree
+      ancestor = prefix.last
+      postfix    <- genNonEmptyBlockTree(ancestor.id)
+      descendant <- Gen.oneOf(postfix)
+      data = TestData(prefix ++ postfix)
```

Comment:
Do we need the ancestors of the `ancestor`? Why not just generate a single block tree?

Reply:
I wanted to make sure that I can pick any 2 blocks in the tree which are related, and that the traversal will only go between them, not higher to the root for example. 

Conceptually I only need a single block tree, but the generator only returns a list of blocks, it's not obvious how to pick an ancestor and a descendant from that, unless one of them is the root. By doing it in 2 stages, I know that every block in the `postfix` is a descendant of the `ancestor`, so I can pick any of them, and `ancestor` may or may not be the `root` of the `prefix`. 

I figured it's easier to generate the data in the most convenient way, so there doesn't need to be any logic in the test.

Reply:
> that the traversal will only go between them, not higher to the root for example.

Makes sense, but we're not really testing that, are we? I mean we're only looking at the result, not at how it's produced. In fact, maybe it could actually test the traversal if the ancestor was always the root, as it would produce an error if we try to access its parent?

Not a big deal, just pointing out

Reply:
For a few minutes I didn't know what you meant by not testing it. Now I understand you meant that it _could_ have accessed the parent during traversal, just not returned it. 

Sorry, I didn't mean to check what the traversal does behind the scenes, only that we return the `ancestor` as the `head` of `path` (which we [check](https://github.com/input-output-hk/metronome/pull/39/files/7fd73c729ad087906359aa78ad637b9ce3b41108#diff-040272a08fe005f889fce95f80d70b6c857e6cf9a776c14a0fd001c98628a9fcR274-R275)), not just by accident because it doesn't have a parent to go to, but because that's what the result is supposed to be. 

The `ancestor` _can_ be the root of the prefix (if it's a single element tree), so if it threw an exception trying to access its parent it should fail the test in some cases. I assume you didn't want to suggest by "always the root" that this should be the only way we should test it?

Reply:
Yeah, makes sense 👍